tests/foreman/cli/test_hammer.py

```diff
@@ -136,6 +136,13 @@ def test_positive_disable_hammer_defaults(request, function_product, target_sat)
 
     :BZ: 1640644, 1368173
     """
+
+    @request.addfinalizer
+    def _finalize():
+        target_sat.cli.Defaults.delete({'param-name': 'organization_id'})
+        result = target_sat.execute('hammer defaults list')
+        assert str(function_product.organization.id) not in result.stdout
+
     target_sat.cli.Defaults.add(
         {'param-name': 'organization_id', 'param-value': function_product.organization.id}
     )
```

Comment:
Yes, Good catch, I've seen this behaviour earlier while writing some test, just one suggestion here, to add finalizer at least after we add entity for better readability
```suggestion
    target_sat.cli.Defaults.add(
        {'param-name': 'organization_id', 'param-value': function_product.organization.id}
    )
    @request.addfinalizer
    def _finalize():
        target_sat.cli.Defaults.delete({'param-name': 'organization_id'})
        result = target_sat.execute('hammer defaults list')
        assert str(function_product.organization.id) not in result.stdout
```

Reply:
@Gauravtalreja1 It can be done for finalizers that touch just one thing, but as long as you change more items, you can't have them all defined before, because it would defer the point of making sure finalizer is always added. This may be odd to read, but on the other hand it also communicates that the finalizer behaves differently from the rest of the test code, meaning it is not executed first but needs to be added as early as possible